plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/mapping/CacheBasedIdentifierMapping.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc.mapping;
+
+import com.google.common.base.CharMatcher;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.jdbc.BaseJdbcClient;
+import io.trino.plugin.jdbc.JdbcIdentity;
+import io.trino.spi.TrinoException;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static io.trino.plugin.jdbc.JdbcErrorCode.JDBC_ERROR;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public final class CacheBasedIdentifierMapping
+        implements IdentifierMapping
+{
+    private final Cache<JdbcIdentity, Map<String, String>> remoteSchemaNames;
+    private final Cache<RemoteTableNameCacheKey, Map<String, String>> remoteTableNames;
+    private final DefaultIdentifierMapping defaultIdentifierMapping;
```

Comment:
name it `delegate`

Reply:
No. `DefaultIdentifierMapping` has specific implementation that I wanted to consciously use here. Hence type is not generic `IdentifierMapping`.